File: backend/voting-app/src/main/java/com/tu/votingapp/utils/mappers/RegionMapper.java
package com.tu.votingapp.utils.mappers;

import com.tu.votingapp.dto.general.RegionDTO;
import com.tu.votingapp.entities.RegionEntity;
import org.mapstruct.Mapper;

@Mapper(componentModel = "spring")
public interface RegionMapper {
    RegionDTO toDto(RegionEntity entity);
    RegionEntity toEntity(RegionDTO dto);
}

File: frontend/android/app/src/main/java/com/example/android/utils/mappers/RegionMapper.kt
package com.example.android.utils.mappers

import com.example.android.dto.response.RegionResponseDTO
import com.example.android.entity.RegionEntity

object RegionMapper {
    fun RegionResponseDTO.toEntity(): RegionEntity = RegionEntity(
        id = this.id,
        name = this.name,
        population = this.population
    )

    fun RegionEntity.toDto(): RegionResponseDTO = RegionResponseDTO(
        id = this.id!!,
        name = this.name,
        population = this.population
    )
}